ext/DimensionalDataAbstractFFTsExt.jl

```diff
@@ -0,0 +1,365 @@
+    else
+        dd
+    end
+
+    _fft_dd = plan.p * reinterpret_into_datatype(T, parent(input_to_plan))
+    fft_dd = reinterpret(typeof(real(oneunit(T)) * oneunit(X) * oneunit(eltype(_fft_dd))), _fft_dd)
+    dd_out = DimArray(fft_dd, lookups)
+    if !is_inverse_transform(plan)
+        correct_phase_reference!(dd_out, plan, lookup(dd), -1)
+    end
+    dd_out
+end
+function LinearAlgebra.mul!(dd_out::DD.AbstractDimArray{Tout, N}, plan::DDPlan{<:Any, Tp}, dd_in::DD.AbstractDimArray{Tin, N}) where {N, Tin, Tout, Tp}
+    lookups_out = get_freqs(plan, dd_in)
+    if !all(i -> dims(dd_out, i) == lookups_out[i], 1:N)
+        throw(ArgumentError("The lookups of the output and input DimensionalData must match."))
+    end
+    input_to_plan = if is_inverse_transform(plan)
+        copyto!(parent(plan.temp), parent(dd_in))
+        correct_phase_reference!(plan.temp, plan, lookups_out, 1)
+        plan.temp
+    else
+        dd_in
+    end
+
+    LinearAlgebra.mul!(reinterpret_into_datatype(Tout, parent(dd_out)), plan.p, reinterpret_into_datatype(Tin, parent(input_to_plan)))
+    if !is_inverse_transform(plan)
+        correct_phase_reference!(dd_out, plan, lookup(dd_in), -1)
+    end
+    
+    # Corrects for the unit of the output. For example, if the input is g*s and the output is kg*s.
+    correct_unit = real(oneunit(Tin) * oneunit(Tp) / oneunit(Tout))
+    dd_out .*= correct_unit # Also used as a check to ensure that the output has the correct units.
+
+    dd_out
+end
+
+"""
+    correct_phase_reference!(dd::DD.AbstractDimArray{<:Any, N}, plan::DDPlan, lookup_refs, phase_scaling) where N = nothing
+
+Multiply the input `dd` by a phase ramp to account to account for the spatial sampling of the input spatial data.
+"""
+function correct_phase_reference!(dd::DD.AbstractDimArray{<:Any, N}, plan::DDPlan, lookup_refs, phase_scaling) where N
+    for i in fftdims(plan)
+        @d dd .*= exp.((im * 2π * phase_scaling * first(lookup_refs[i])) .* dims(dd, i))
+    end
+end
+
+for i in (:fft, :ifft, :rfft)
+    signal = i == :ifft ? "i" : "-i"
+    fft_freq = i == :ifft ? "ifftfreq" : "fftfreq"
+    docsstring = """
+          fft(A [, dims])
+        
+        Performs a multidimensional $(i) of the array `A`. The optional `dims` argument specifies an iterable subset of dimensions (e.g. an integer, range, tuple, or array) to transform along. A one-dimensional $(i) computes the one-dimensional discrete Fourier transform (DFT) as defined by:
+        
+        ```math
+        \\operatorname{DFT}(A)[k] =
+        \\sum_{n=1}^{\\operatorname{length}(A)} \\exp\\left($(signal)\\pi x[n]f[k] \\right) A[n] dx.
+        ```
+        where `x` is the lookup dimension of the input data, `f` is the lookup dimension of the output data, and `dx` is the step of the lookup dimension of the input data.
+
+        Return a DimensionalData array with the transformed data, where the lookup dimensions are the output of $(fft_freq)
+        """
+    eval(quote
```

Comment:
```suggestion
    @eval begin
```

This syntax is more often used

Reply:
Implemented these changes on 6ff6b8d